### PR TITLE
Support for opening multiple sqlite databases

### LINKED ISSUE #126: Cannot use multiple databases at once
I would like to be able to use DatabaseA while importing DatabaseB (so my app doesn't have any period of downtime where the database is in an invalid state).

## Issue

**At the moment, passing a different sqlitePath results in the library silently ignoring the path and overwriting the old database.**

This is because `openDb()` ignores all config if there is an existing database connection.

[import.js](https://github.com/BlinkTagInc/node-gtfs/blob/master/lib/import.js#L356)
```js
const importGtfs = async (initialConfig) => {
  ...
  const db = await openDb(config)
  ...
}
```

[db.js](https://github.com/BlinkTagInc/node-gtfs/blob/184d69af6c01e3811f3a91867564cd620ed40c35/lib/db.js#L6)
```js
export async function openDb(initialConfig) {
  const config = setDefaultConfig(initialConfig);
  if (!db) {
    db = await open({
      filename: config.sqlitePath,
      driver: sqlite3.Database,
    });
  }

  return db;
}
```

Relevant issue #117.

## Background info

My use case involves importing data from multiple agencies (each of which is updated separately so I need to be able to update one without DROP TABLEing all the others). This could be done by using multiple database files. I think it would be ideal if there was an additional (optional) parameter for library functions to specify the database (see [my branch](https://github.com/DarkMatterMatt/node-gtfs/tree/feat/multiple-databases)). Query functions like `getAgencies()` would have a fourth parameter for the database (defaults to current behaviour, fetching the database using `getDb()`). and `importGtfs()` would look more like:
```js
const importGtfs = async (initialConfig, db) => {
  ...
  db = db ?? await openDb(config)
  ...
}
```

Thanks :)

### PR DESCRIPTION
**Pull request by brendannee**
Fixes #126

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -1,36 +1,64 @@
...
-  db = undefined;
+  delete dbs[filename];
 }
 
-export function getDb() {
-  if (db) {
-    return db;
+export function getDb(config) {
+  if (config) {
+    return dbs[config.sqlitePath];
```
Comment (DarkMatterMatt) [inline lib/db.js:68]: If config is supplied, but `sqlitePath` is not set, then it should probably default to the `:memory:` database to be consistent with `openDb()`?

File: lib/db.js [hunk 2]
```diff
@@ -1,36 +1,76 @@
...
+  }
+
+  if (!config && Object.keys(dbs).length === 1) {
```
Comment (DarkMatterMatt) [inline lib/db.js:71]: `!config` is always true

File: lib/db.js [hunk 3]
```diff
@@ -1,36 +1,76 @@
...
+    const filename = Object.keys(dbs)[0];
+    return dbs[filename];
   }
 
   throw new Error(
```
Comment (DarkMatterMatt) [inline lib/db.js:76]: This will also be thrown if there is more than one open connection (incorrect error message)

Comment (DarkMatterMatt) [inline lib/db.js:76]: This will also be thrown if there is more than one open connection (incorrect error message)

File: lib/db.js (hunk 1 above)
Comment (brendannee) [reply lib/db.js:75]: Great point. I updated it to work like openDb and so will default to :memory:

**Comment by brendannee** [comment]
Logic is adjusted - thanks for the feedback. Now it will use the database you specify in config if it is open, otherwise it will fall back to whatever database is already open if you only have one open (which should ensure backwards compatibility).

Let me know what you think and if you see any other issues.

**Comment by DarkMatterMatt** [comment]
Looks great!

**Pull request merged**